tests/tests.rs

```diff
@@ -456,9 +456,8 @@ fn test_dynamic_roots() {
 #[test]
 #[should_panic]
 fn test_dynamic_bad_set() {
-    let arena1: Arena<Rootable![DynamicRootSet<'_>]> = Arena::new(|mc| DynamicRootSet::new(mc));
-
-    let arena2: Arena<Rootable![DynamicRootSet<'_>]> = Arena::new(|mc| DynamicRootSet::new(mc));
```

Comment:
One thing I noticed here is that `let arena: Arena<Rootable![MyRoot<'_>]> = Arena::new(|mc| ...);` no longer works, but `let arena = Arena::<Rootable![MyRoot<'_>]>::new(|mc| ...);` does, and I have no idea why.

It's a very minor paper cut, but it'd be nice not to break it.